src/wrapper.rs

```diff
@@ -209,7 +222,22 @@ impl Scalar {
             )),
         }
     }
+    fn __int__(&self) -> BigUint {
+        // Bug, Fr::to_string will print nothing if the value is zero
+        BigUint::from_str(&*self.0.to_string()).unwrap_or(BigUint::ZERO)
+    }
 
+    fn pow(&self, exp: Scalar) -> PyResult<Scalar> {
+        let bls_modulus = BigUint::from_str(BLS_MODULUS).unwrap();
+        let base_bigint = BigUint::from_bytes_le(self.to_le_bytes()?.as_slice());
+        let exp_bigint = BigUint::from_bytes_le(exp.to_le_bytes()?.as_slice());
+        let result = base_bigint.modpow(&exp_bigint, &bls_modulus);
+        Ok(Scalar(
+            ark_bls12_381::Fr::from_str(&*result.to_string()).map_err(|_| {
+                exceptions::PyValueError::new_err("Failed to convert result to scalar")
+            })?,
+        ))
+    }
```

Comment:
Not blocking as we can speed this up after -- Fr should have a pow method that one can use too, which should allow us to remove the BLS_MODULUS import

Reply:
I think its more than a u32, last time I checked or that would be almost unusable!

Reply:
It might have been &[u32] or something of that nature

Reply:
Ah you're right. I can fix this! Here's a screenshot of the method:

<img width="637" alt="image" src="https://github.com/user-attachments/assets/f0cd78f4-be98-4557-9576-a5e3e2657478">

Reply:
No worries! Feel free to push another PR when readyb :)

Reply:
Made this PR:

* https://github.com/crate-crypto/py-arkworks-bls12381/pull/7